### PR TITLE
Resolve #3: Implement Join Group/ Browse Groups view

### LINKED ISSUE #3: Implement Join Group/ Browse Groups view

### PR DESCRIPTION
**Pull request by ckucera3**

### CODE REVIEW

File: src/popup/groups/groups.scss [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  width: 40px;
+  padding: 7px;
+}
+.up-btn:hover {
+  background-color: rgba(0, 0, 0, .1);
+}
+.up-btn:active {
+  background-color: rgba(0, 0, 0, .2);
+}
+
```
Comment (Bkucera) [inline src/popup/groups/groups.scss]: I would remove everything above with .up-btn and :hover :active, and replace it all with just:
```javascript
.up-btn {
    position: absolute;
    top:0; right:0;
}
```

File: src/popup/groups/groups.html [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<div class="groups-section">
+    <button class="icon-btn up-btn">
+        <i class="material-icons">arrow_upwards</i>
```
Comment (Bkucera) [inline src/popup/groups/groups.html]: this should be `arrow_upward` instead of upwards

File: src/model/User.js [hunk 3]
```diff
@@ -44,7 +44,18 @@ export function getCurrentUser() {
...
+function getUserDBRef() {
+    const db = firebase.database().ref().child('users').child(getCurrentUser().uid)
+    return db
+}
+
+
+export function joinGroup(groupId) {
+    const userRef = getUserDBRef()
+    return userRef.child('groups').child(groupId).set(true)
+
```
Comment (Bkucera) [inline src/model/User.js]: what about setting "groups/members/:groupId/:uid" to true ?

File: src/model/User.js [hunk 4]
```diff
@@ -44,7 +44,18 @@ export function getCurrentUser() {
   return firebase.auth().currentUser
 }
 
-window.getGroups = getMemberGroups
+function getUserDBRef() {
+    const db = firebase.database().ref().child('users').child(getCurrentUser().uid)
+    return db
+}
```
Comment (Bkucera) [inline src/model/User.js]: Don't think we need this. Its more readable to me to just access `ref('users').child(getCurrentUser().uid)` when its needed.

File: src/popup/groups/groups.scss (hunk 1 above)
Comment (ckucera3) [inline src/popup/groups/groups.scss]: I agree.

Comment (ckucera3) [inline src/popup/groups/groups.scss]: Done

**Comment by ckucera3** [comment]
Requested changes have been addressed.

**Pull request merged**